actors/runtime/src/main/java/cloud/orbit/actors/runtime/KryoSerializer.java

```diff
@@ -376,8 +469,50 @@ public void serializeMessage(BasicRuntime basicRuntime, OutputStream outputStrea
                 writeNodeAddress(out, message.getReferenceAddress());
                 out.writeInt(message.getInterfaceId());
                 out.writeInt(message.getMethodId());
-                kryo.writeClassAndObject(out, message.getObjectId());
-                kryo.writeObjectOrNull(out, message.getHeaders(), HashMap.class);
+
+                ValueType valueTypeForObjectId = ValueType.getType(message.getObjectId());
+                out.writeByte(valueTypeForObjectId.id);
+                if (valueTypeForObjectId.equals(ValueType.STRING))
+                {
+                    out.writeString(String.valueOf(message.getObjectId()));
+                }
+                else if (valueTypeForObjectId.equals(ValueType.INT))
+                {
+                    out.writeInt((Integer) message.getObjectId());
+                }
+                else
+                {
+                    kryo.writeClassAndObject(out, message.getObjectId());
+                }
+
+                Map<Object, Object> headers = message.getHeaders();
+                if (headers != null)
+                {
+                    out.writeInt(headers.size());
+                    for (Map.Entry<Object, Object> entry : headers.entrySet())
+                    {
+                        out.writeString(String.valueOf(entry.getKey()));
```

Comment:
@JoeHegarty this also means that the key for sticky headers must be string, we might want to do some type checks or the same value type magic as per below, wdyt?